src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -16,8 +16,8 @@
 
 	Optional<User> findFirstByUserName(@NotBlank String userName);
 
-	@Query("select u from User u where u.birthDate between :dateFrom and :dateTo and u.name like :cadena")
+	@Query("select u from User u where ((cast(:dateFrom as date) is null or cast(:dateTo as date) = null or"
+	        + " u.birthDate between :dateFrom and :dateTo) and (:name = null or lower(u.name) like lower(concat('%',:name,'%'))))")
```

Comment:
Los = null deberían ir con **"is null"** porque con = null puede haber motores de bases de datos que no lo interpreten y devuelvan resultados inesperados.